drakrun/drakrun/draksetup.py

```diff
@@ -245,6 +246,86 @@ def send_usage_report(report):
         logging.exception("Failed to send usage report. This is not a serious problem.")
 
 
+def create_rekall_profiles(install_info):
+    profiles_path = os.path.join(LIB_DIR, "profiles")
+
+    with tempfile.TemporaryDirectory() as mount_path:
+        # we mount 2nd partition, as 1st partition is windows boot related and 2nd partition is C:\\
+
+        if install_info["storage_backend"] == "zfs":
+            # workaround for not being able to mount a snapshot
+            base_snap = shlex.quote(os.path.join(install_info["zfs_tank_name"], 'vm-0@booted'))
+            tmp_snap = shlex.quote(os.path.join(install_info["zfs_tank_name"], 'tmp'))
+            try:
+                subprocess.check_output(f'zfs clone {base_snap} {tmp_snap}', shell=True)
+            except subprocess.CalledProcessError:
+                logging.warning("Failed to clone temporary zfs snapshot. Aborting generation of usermode rekall profiles")
+                return
+
+            tmp_mount = shlex.quote(os.path.join("/", "dev", "zvol", install_info["zfs_tank_name"], "tmp-part2"))
+            try:
+                # We have to wait for a moment for zvol to appear
+                time.sleep(1.0)
+                subprocess.check_output(f'mount -t ntfs -o ro {tmp_mount} {mount_path}', shell=True)
+            except subprocess.CalledProcessError:
+                logging.warning("Failed to mount temporary zfs snapshot. Aborting generation of usermode rekall profiles")
+                try:
+                    subprocess.check_output(f'zfs destroy {tmp_snap}', shell=True)
+                except subprocess.CalledProcessError:
+                    logging.exception('Failed to cleanup after zfs tmp snapshot')
+                return
+        else:  # qcow2
+            try:
+                subprocess.check_output("modprobe nbd", shell=True)
+            except subprocess.CalledProcessError:
+                logging.warning("Failed to load nbd kernel module. Aborting generation of usermode rekall profiles")
+                return
+
+            # TODO: this assumes /dev/nbd0 is free
+            try:
+                subprocess.check_output(f"qemu-nbd -c /dev/nbd0 --read-only {os.path.join(LIB_DIR, 'volumes', 'vm-0.img')}", shell=True)
```

Comment:
Maybe we could convert qcow2 to raw disk image using
```
qemu-img convert root.qcow2 root.raw
```

and mount it using regular loop device? This would take 2x disk space but I guess might be more reliable.

Reply:
Why are you assuming this method is unreliable? 2x disk space sounds like a horrible idea.

Reply:
> more reliable.

I don't agree. This would put a lot of load on the filesystem, especially if we're considering 100-200G volumes. Writing this amount of data seems unreasonable.

The only issue is that we're trying to mount a dirty NTFS filesystem, but if that's good enough for ntfs-3g it's good enough for us.

Reply:
Ok, that's reasonable. nvm